sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -382,18 +382,11 @@ private case class PostgresDialect()
       case Types.ARRAY =>
         val tableName = rsmd.getTableName(columnIdx)
         val columnName = rsmd.getColumnName(columnIdx)
-        val query =
-          s"""
-             |SELECT pg_attribute.attndims
-             |FROM pg_attribute
-             |  JOIN pg_class ON pg_attribute.attrelid = pg_class.oid
-             |  JOIN pg_namespace ON pg_class.relnamespace = pg_namespace.oid
-             |WHERE pg_class.relname = '$tableName' and pg_attribute.attname = '$columnName'
-             |""".stripMargin
+        val query = s"SELECT array_ndims($columnName) FROM $tableName LIMIT 1"
```

Comment:
Lets add comment

```
// Spark does not support different array dimensionality per row. With this in mind, get dimensionality of any row.
// This might fail down the road, as Postgres array column might contain different dimensionality.
```